src/v/storage/probe.cc

```diff
@@ -261,6 +256,34 @@ void probe::setup_metrics(const model::ntp& ntp) {
       });
 }
 
+void probe::setup_public_metrics(const model::ntp& ntp) {
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    namespace sm = ss::metrics;
+    auto labels = {
+      metrics::namespace_label(ntp.ns()),
+      metrics::topic_label(ntp.tp.topic()),
+      metrics::partition_label(ntp.tp.partition()),
+    };
+
+    auto aggregate_labels = {sm::shard_label, metrics::partition_label};
```

Comment:
We shouldn't have topic labels on public metrics so need to aggregate that as well.